src/classes/ERR_DeleteOutdated.cls

```diff
@@ -0,0 +1,42 @@
+/*
+    Copyright (c) 2018 Salesforce.org
+    All rights reserved.
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @group ErrorHandling
+* @description A queueable job that detects errors which are more than 90 days old and delete them
+*/
+public virtual class ERR_DeleteOutdated implements Queueable {
+    /**
+    * @description Execute method to detect errors more than 90 days old and delete
+    */
+    public void execute(QueueableContext qc) {
+        DateTime now = DateTime.now();
+        DateTime ninetyDaysAgo = now.addDays(-90);
+        List<Error__c> errors = [SELECT Id FROM Error__c WHERE Datetime__c < :ninetyDaysAgo];
```

Comment:
Would this fail if it returned more than 50000 records? In my head we would make this process Batchable instead of Queueable, though I'm not sure about the risks. We could also just limit the query, and eventually it would catch up. @ceiroa Your thoughts?

Reply:
Good point, @lparrott. I've always used a Schedulable that calls a batch instead Queueable. What are the advantages of using Queueable, @xinnanlisf? How do the query limit applies in this case?

Reply:
Hey @lparrott @ceiroa,

Good point! I will change it to a batch class.